src/mpfao/tdympfao_snes.c

```diff
@@ -43,7 +43,7 @@ PetscErrorCode TDyMPFAOSNESPreSolve(TDy tdy) {
 
   // Update the auxillary variables
   ierr = VecGetArray(tdy->soln_prev,&p); CHKERRQ(ierr);
-  ierr = TDyUpdateState(tdy, p); CHKERRQ(ierr);
+  ierr = TDyUpdateState(tdy, p, mesh->num_cells_local); CHKERRQ(ierr);
   ierr = VecRestoreArray(tdy->soln_prev,&p); CHKERRQ(ierr);
```

Comment:
@jeff-cohere You have a good point regarding avoiding the need of passing the third argument i.e.`mesh->num_cells_local`. I will change it by passing `TDyUpdateState` the `Vec` on which the function needs to perform computation.

Reply:
The proposed refactor is causing WY to fail. The refactored branch is `bishtgautam/fix-parallel-runs-rebase-02-02-2022-updatestate`. I haven't been able to debug the error. Can we merge this PR and open a new issue to track this review comment?